composer/loggers/slack_logger.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""Logs metrics to slack using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`."""
+
+from typing import Any, Callable, Dict, Optional
+
+from slack_sdk import WebhookClient
+
+from composer.loggers.logger_destination import LoggerDestination
+
+
+class SlackLogger(LoggerDestination):
+    """Log metrics to slack, using `Slack Webhook URL <https://api.slack.com/messaging/webhooks>`.
+
+    .. note::
+        See example instantiating of SlackLogger in `examples/slack_log_metrics.py`
+        Webhook URL will be passed as an environment variable.
+        Formatter function can be defined at callsite or default formatter function will be applied if not provided.
+
+    Args:
+        webhook_url (str): Slack Webhook URL
+        log_metrics_formatter_func ((...) -> Any | None): A formatter function that returns list of blocks to be sent to slack
+    """
+
+    def __init__(
+        self,
+        webhook_url: str = '',
+        log_metrics_formatter_func: Optional[Callable] = None,
```

Comment:
should we add a log_interval here, where the user can specify how often to log a metric to slack

Reply:
let me do that, I think I could add it in the formatter function as well. Ideally have one param to be one configuration with both formatting + interval

Reply:
I will defer this to another iteration, interval is a little harder to code up. Instead ill have an include_keys config that allows us to log only essential keys, so we minimize the log spam